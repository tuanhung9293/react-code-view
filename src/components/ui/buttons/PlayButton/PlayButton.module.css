@value colors: 'styles/colors.css';
@value white, nasa-yellow from colors;

.wrapper {
    width: 45px;
    cursor: pointer;
}
  
.svg {
    transition: .5s; 
    fill:#7c7c7c; 
    opacity: .5;
}

.wrapper:hover .svg {
    fill: nasa-yellow;
    opacity: 1;
}
