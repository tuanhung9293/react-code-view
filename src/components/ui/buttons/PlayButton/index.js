import React from 'react'
import {Link} from 'react-router-dom'
import styles from './PlayButton.module.css'

function PlayButton({to, ...rest}) {
  return (
    <Link to={to} {...rest}>
      <div className={styles.wrapper}>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 26 26">
          <polygon className={styles.svg} points="9.33 6.69 9.33 19.39 19.3 13.04 9.33 6.69" />
          <path className={styles.svg} d="M26,13A13,13,0,1,1,13,0,13,13,0,0,1,26,13ZM13,2.18A10.89,10.89,0,1,0,23.84,13.06,10.89,10.89,0,0,0,13,2.18Z" />
        </svg>
      </div>
    </Link>
  )
}

export default PlayButton
