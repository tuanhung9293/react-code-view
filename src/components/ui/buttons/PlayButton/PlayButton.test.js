import React from 'react'
import {shallow} from 'enzyme'
import PlayButton from './'

it('renders PlayButton with props to', () => {
  expect(shallow(
    <PlayButton to="/collection" />
  )).toMatchSnapshot()
})

it('PlayButton should pass props down', () => {
  expect(shallow(
    <PlayButton to="/collection" data="data1" data2="data2" />
  )).toMatchSnapshot()
})
