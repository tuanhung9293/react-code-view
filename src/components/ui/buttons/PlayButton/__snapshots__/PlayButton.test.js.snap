// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlayButton should pass props down 1`] = `
<Link
  data="data1"
  data2="data2"
  replace={false}
  to="/collection"
>
  <div
    className="wrapper"
  >
    <svg
      viewBox="0 0 26 26"
      xmlns="http://www.w3.org/2000/svg"
    >
      <polygon
        className="svg"
        points="9.33 6.69 9.33 19.39 19.3 13.04 9.33 6.69"
      />
      <path
        className="svg"
        d="M26,13A13,13,0,1,1,13,0,13,13,0,0,1,26,13ZM13,2.18A10.89,10.89,0,1,0,23.84,13.06,10.89,10.89,0,0,0,13,2.18Z"
      />
    </svg>
  </div>
</Link>
`;

exports[`renders PlayButton with props to 1`] = `
<Link
  replace={false}
  to="/collection"
>
  <div
    className="wrapper"
  >
    <svg
      viewBox="0 0 26 26"
      xmlns="http://www.w3.org/2000/svg"
    >
      <polygon
        className="svg"
        points="9.33 6.69 9.33 19.39 19.3 13.04 9.33 6.69"
      />
      <path
        className="svg"
        d="M26,13A13,13,0,1,1,13,0,13,13,0,0,1,26,13ZM13,2.18A10.89,10.89,0,1,0,23.84,13.06,10.89,10.89,0,0,0,13,2.18Z"
      />
    </svg>
  </div>
</Link>
`;
